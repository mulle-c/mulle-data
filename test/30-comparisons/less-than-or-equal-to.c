#include <mulle-data/mulle-data.h>

#include <stdio.h>

static void  mulle_range_print( struct mulle_range a)
{
   if( ! a.length)
      printf( "NULL");  
   else
      printf( "%tu-%tu (%tu)", a.location, a.location + a.length - 1, a.length);
}


int   main( int argc, char *argv[])
{
   struct mulle_range   a;
   uintptr_t            location;

   for( a.location = 0; a.location < 4; a.location += 1)
      for( a.length = 0; a.length < 4; a.length += 1)
         for( location = 0; location <= 6; location += 1)
         {
            mulle_range_print( a);
            printf( " <= %tu = %s\n",
               location,
               mulle_range_less_than_or_equal_to_location( a, location) ? "YES" : "NO");
         }

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
