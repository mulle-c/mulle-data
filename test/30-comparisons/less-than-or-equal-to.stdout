NULL <= 0 = NO
NULL <= 1 = NO
NULL <= 2 = NO
NULL <= 3 = NO
NULL <= 4 = NO
NULL <= 5 = NO
NULL <= 6 = NO
0-0 (1) <= 0 = YES
0-0 (1) <= 1 = YES
0-0 (1) <= 2 = YES
0-0 (1) <= 3 = YES
0-0 (1) <= 4 = YES
0-0 (1) <= 5 = YES
0-0 (1) <= 6 = YES
0-1 (2) <= 0 = NO
0-1 (2) <= 1 = YES
0-1 (2) <= 2 = YES
0-1 (2) <= 3 = YES
0-1 (2) <= 4 = YES
0-1 (2) <= 5 = YES
0-1 (2) <= 6 = YES
0-2 (3) <= 0 = NO
0-2 (3) <= 1 = NO
0-2 (3) <= 2 = YES
0-2 (3) <= 3 = YES
0-2 (3) <= 4 = YES
0-2 (3) <= 5 = YES
0-2 (3) <= 6 = YES
NULL <= 0 = NO
NULL <= 1 = NO
NULL <= 2 = NO
NULL <= 3 = NO
NULL <= 4 = NO
NULL <= 5 = NO
NULL <= 6 = NO
1-1 (1) <= 0 = NO
1-1 (1) <= 1 = YES
1-1 (1) <= 2 = YES
1-1 (1) <= 3 = YES
1-1 (1) <= 4 = YES
1-1 (1) <= 5 = YES
1-1 (1) <= 6 = YES
1-2 (2) <= 0 = NO
1-2 (2) <= 1 = NO
1-2 (2) <= 2 = YES
1-2 (2) <= 3 = YES
1-2 (2) <= 4 = YES
1-2 (2) <= 5 = YES
1-2 (2) <= 6 = YES
1-3 (3) <= 0 = NO
1-3 (3) <= 1 = NO
1-3 (3) <= 2 = NO
1-3 (3) <= 3 = YES
1-3 (3) <= 4 = YES
1-3 (3) <= 5 = YES
1-3 (3) <= 6 = YES
NULL <= 0 = NO
NULL <= 1 = NO
NULL <= 2 = NO
NULL <= 3 = NO
NULL <= 4 = NO
NULL <= 5 = NO
NULL <= 6 = NO
2-2 (1) <= 0 = NO
2-2 (1) <= 1 = NO
2-2 (1) <= 2 = YES
2-2 (1) <= 3 = YES
2-2 (1) <= 4 = YES
2-2 (1) <= 5 = YES
2-2 (1) <= 6 = YES
2-3 (2) <= 0 = NO
2-3 (2) <= 1 = NO
2-3 (2) <= 2 = NO
2-3 (2) <= 3 = YES
2-3 (2) <= 4 = YES
2-3 (2) <= 5 = YES
2-3 (2) <= 6 = YES
2-4 (3) <= 0 = NO
2-4 (3) <= 1 = NO
2-4 (3) <= 2 = NO
2-4 (3) <= 3 = NO
2-4 (3) <= 4 = YES
2-4 (3) <= 5 = YES
2-4 (3) <= 6 = YES
NULL <= 0 = NO
NULL <= 1 = NO
NULL <= 2 = NO
NULL <= 3 = NO
NULL <= 4 = NO
NULL <= 5 = NO
NULL <= 6 = NO
3-3 (1) <= 0 = NO
3-3 (1) <= 1 = NO
3-3 (1) <= 2 = NO
3-3 (1) <= 3 = YES
3-3 (1) <= 4 = YES
3-3 (1) <= 5 = YES
3-3 (1) <= 6 = YES
3-4 (2) <= 0 = NO
3-4 (2) <= 1 = NO
3-4 (2) <= 2 = NO
3-4 (2) <= 3 = NO
3-4 (2) <= 4 = YES
3-4 (2) <= 5 = YES
3-4 (2) <= 6 = YES
3-5 (3) <= 0 = NO
3-5 (3) <= 1 = NO
3-5 (3) <= 2 = NO
3-5 (3) <= 3 = NO
3-5 (3) <= 4 = NO
3-5 (3) <= 5 = YES
3-5 (3) <= 6 = YES
